tracking-migration/src/main/java/io/customer/tracking/migration/MigrationAssistant.kt

```diff
@@ -55,6 +57,16 @@ class MigrationAssistant private constructor(
         }
     }
 
+    private fun migrateExistingDevice() {
+        // If there is no old device token, then either the device was never added or it was already migrated
+        val oldDeviceIdentifier = sitePreferences.getDeviceToken() ?: return
+
+        logger.debug("Migrating existing device with token: $oldDeviceIdentifier")
+        globalPreferenceStore.saveDeviceToken(oldDeviceIdentifier)
```

Comment:
While this should work fine, but storing token in new `globalPreferenceStore` should be responsibility of `MigrationProcessor` and not `MigrationAssistant`. `MigrationAssistant` is currently responsible for fetching and removing existing data only. Asking it to process and save in new store is violating the responsibility in my opinion and this should be handed over to `MigrationProcessor` instead.